### DISCUSSION TITLE
Generating types with @supabase/auth-helpers-sveltekit

### DISCUSSION THREAD

**Discussion by Etheonor** [supabase/supabase#11811] [Questions]
Hello,

I'm not sure if I made a mistake or if it's not available. 

I used this page to generate my DB types and everything went well https://supabase.com/docs/guides/api/generating-types

But when I wanted to add it on my supabaseClient, I had an error ('Expected 0 type arguments, but got 1.'). It seems to be an issue with the createClient function from @supabase/auth-helpers-sveltekit because if I try to use createClient from supabase-js, it seems to work. 

Is there a solution or do I need to create an issue on the auth-helpers-sveltekit repo?

![image](https://user-images.githubusercontent.com/35852770/213747237-a7450578-e9dd-4821-ab59-b16b3433516d.png)

**Comment by kasperlaursen** [comment]
Hi @Etheonor.

I had the same issue as you here, but I managed to find the solution! 😁

The return type of the SupabaseClient from the Auth helper is `TypedSupabaseClient`. 
It gets the type from the `app.d.ts` file. 

This is an example of mine, where i generated the types to the `src/lib/data/supabase` folder in a file called `types.ts`.  
Using this command:   
```shell
npx supabase gen types typescript --project-id 'PTOJECT_ID' --schema public > src/lib/data/supabase/types.ts
```
My `app.d.ts` file:
```ts
// src/app.d.ts
declare namespace App {
	interface Supabase {
		Database: import('./lib/data/supabase/types').Database;
		SchemaName: 'public';
	}
	// interface Error {}
	// interface Locals {}
	// interface PageData {}
	// interface Platform {}
}
``

**Comment by mmmmillar** [comment]
Hi @kasperlaursen, is there anything you had to run after updating `app.d.ts`? I'm still having the issue:

```
> app@0.0.1 check
> svelte-kit sync && svelte-check --tsconfig ./tsconfig.json

====================================
Loading svelte-check in workspace
Getting Svelte diagnostics...

/lib/db.ts:5:38
Error: Expected 0 type arguments, but got 1. 

export const supabase = createClient<Database>(PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY);
```

```
// app.d.ts
declare global {
	namespace App {
		// interface Error {}
		// interface Locals {}
		interface Supabase {
			Database: import('./lib/types').Database;
			SchemaName: 'public';
		}
		interface PageData {
			session: import('@supabase/auth-helpers-sveltekit').SupabaseSession;
		}
		// interface Platform {}
	}
}

export {};
```

```
// types.ts
export interface Database {
  public: {
    Tables: {
      projects: {
        Row: {
          artist_id: string | null
          cost: number
          created_at: string | null
          id: string
          is_funded: boolean
          name: string
          summary: string
```

**Comment by PeDeRato** [comment]
same issue here, still don't know how to fix it :(
